src/AppBundle/Resources/views/Restaurant/index.html.twig

```diff
@@ -100,9 +100,29 @@
   </div>
 </div>
 
+<div id='cart-warning-modal' class="modal fade" tabindex="-1" role="dialog">
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+        <h4 class="modal-title">{{'cart.modal.title'|trans}}</h4>
+      </div>
+      <div class="modal-body">
+        <p>{{'cart.modal.content'|trans}}</p>
+      </div>
+      <div class="modal-footer">
+        <button id='cart-warning-default' type="button" class="btn btn-default" data-dismiss="modal">{{'cart.modal.cancel'|trans }}</button>
+        <button id='cart-warning-primary' type="button" class="btn btn-primary">{{ 'cart.modal.confirm'|trans}}</button>
+      </div>
+    </div>
+  </div>
+</div>
+</div>
+
 {% endblock %}
 
 {% block scripts %}
+{% set sessionCart = app.session.get('cart') %}
```

Comment:
This is not useful as the `cart` variable in the template [is the session cart](https://github.com/coopcycle/coopcycle-web/blob/2f547595a6a5bf1d0de89cb99b2b8a2f32658a4d/src/AppBundle/Controller/RestaurantController.php#L233)

Reply:
I think it is. The cart variable from the template will be erased if the 'old' session cart is from a different restaurant. 
To be able to trigger the modal, and ask the user to choose if he want to continue with the old cart or create a new one we need to keep the old one.